contracts/transfer/src/transfer.rs

```diff
@@ -0,0 +1,132 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use core::convert::TryFrom;
+
+use alloc::vec::Vec;
+use canonical::{Canon, Store};
+use canonical_derive::Canon;
+use dusk_bls12_381::BlsScalar;
+use dusk_bytes::Serializable;
+use dusk_jubjub::JubJubAffine;
+use dusk_kelvin_map::Map;
+use dusk_pki::PublicKey;
+use phoenix_core::{Crossover, Fee, Message, Note};
+
+mod tree;
+use tree::Tree;
+
+#[derive(Debug, Default, Clone, PartialEq, Eq, PartialOrd, Ord, Canon)]
+pub struct PublicKeyBytes([u8; PublicKey::SIZE]);
+
+impl From<PublicKey> for PublicKeyBytes {
+    fn from(pk: PublicKey) -> Self {
+        Self(pk.to_bytes())
+    }
+}
+
+#[derive(Debug, Default, Clone, Canon)]
+pub struct Transfer<S: Store> {
```

Comment:
I'd rename to `TransferContract` to avoid any missconceptions with the methods. Maybe just `Contract` and allow the user to alias it if needed.

Reply:
Agreed, fixed